modules/post/multi/gather/enum_software_versions.rb

```diff
@@ -0,0 +1,124 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Multiplatform Installed Software Version Enumerator',
+        'Description' => %q{
+          This module, when run against a compromised machine, will gather details on all installed software,
+          including their versions and if available, when they were installed, and will save it into a loot file for later use.
+          Users can then use this loot file to determine what additional vulnerabilites may affect the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'gwillcox-r7' ],
+        'Platform' => %w[win linux osx bsd solaris android],
+        'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+    )
+  end
+
+  def store_linux_loot(listing)
+    file = store_loot('host.linux.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+    print_good("Stored information about the installed products to the loot file at #{file}")
+  end
+
+  # Run Method for when run command is issued
+  def run
+    case session.platform
+    when 'windows'
+      if command_exists?('wmic') == false
+        print_error("The 'wmic' command doesn't exist on this host!") # wmic is technically marked as depreciated so this command could very well be removed in future releases.
+        return
+      end
+      listing = cmd_exec('wmic product get Name, Description, Version, InstallDate', nil, 6000).to_s
+      unless listing.include?('Description')
+        print_error('Was unable to get a listing of installed products...')
+        return
+      end
+      file = store_loot('host.windows.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+      print_good("Stored information about the installed products to the loot file at #{file}")
+    when 'linux'
+      # All of the following options were taken from https://distrowatch.com/dwres.php?resource=package-management
+      cmd = %w[hostnamectl]
+      if command_exists?('hostnamectl') == false
```

Comment:
Not sure if it is important for this module, but this command is systemd specific and won't be available on Gentoo, Apline, Artix, etc.

Reply:
That would work, too. You could then add programming-language specific ones, too, like `gem`, `pip`, and friends.

Reply:
@acammack-r7 To reduce repeating myself, I'll put my answers to this problem in the thread above where I am talking to bcoles about this. Nutshell version here is that I think the way we are doing things is still fine given that everything I'm seeing online suggests users should not install multiple package managers and even Arch who is known for customization warns against doing this.

Furthermore the ones that do do this tend to just be wrappers around existing package managers, rather than full replacements per say.

Also I think we can check some more files in addition to this command to get things working.

Reply:
@acammack-r7 Decided to go with an approach that does not use this command in its strategy. Marking this as resolved for now. Feel free to reopen if needed.

Reply:
@bcoles Thanks, will read up on that now; I saw there was quite a bit of controversy surrounding it which was a part of the reason why I decided to move away from my previous approach